compiler/rustc_codegen_llvm/src/back/lto.rs

```diff
@@ -696,6 +699,8 @@ pub unsafe fn optimize_thin_module(
     let module_name = &thin_module.shared.module_names[thin_module.idx];
     let tm_factory_config = TargetMachineFactoryConfig::new(cgcx, module_name.to_str().unwrap());
     let tm = (cgcx.tm_factory)(tm_factory_config).map_err(|e| write::llvm_err(&diag_handler, e))?;
+    let tm_cpp_args = create_target_machine_args(&cgcx.expanded_args);
+    add_cpp_args_to_target_machine(tm, &*tm_cpp_args);
```

Comment:
If I'd want to do it during the creation I'd need to change this type
https://github.com/rust-lang/rust/blob/1.71.1/compiler/rustc_codegen_llvm/src/lib.rs#L175
to a tuple of TargetMachine & TargetmachineCppArgs
so that TargetMachineFactoryFn has the correct type signiture
https://github.com/rust-lang/rust/blob/1.71.1/compiler/rustc_codegen_ssa/src/back/write.rs#L308-L316

I fear this change might be to massive.
If I am overlooking something and this can be easily done please tell me :)

Reply:
I think changing the factory signature should be fine.
(But you can save the current version of the PR to a branch for easy revert, just in case.)

Reply:
I'll try that then.